dask/dataframe/io/tests/test_sql.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import print_function, division, absolute_import
+
+import io
+import pytest
+from dask.utils import tmpfile
+from dask.dataframe.io.sql import read_sql_table
+pd = pytest.importorskip('pandas')
+dd = pytest.importorskip('dask.dataframe')
+pytest.importorskip('sqlalchemy')
+pytest.importorskip('sqlite3')
+
+
+data = """
+name,number,age,negish
+Alice,0,33,-5
+Bob,1,40,-3
+Chris,2,22,3
+Dora,3,16,5
+Edith,4,53,0
+Francis,5,30,0
+Garreth,6,20,0
+"""
+
+df = pd.read_csv(io.StringIO(data), index_col='number')
+
+
+@pytest.yield_fixture
+def db():
+    with tmpfile() as f:
+        uri = 'sqlite:///%s' % f
+        df.to_sql('test', uri, index=True, if_exists='replace')
+        yield uri
+
+
+def test_simple(db):
+    # single chunk
+    data = read_sql_table('test', db, partitions=2, index_col='number').compute()
+    assert (data.name == df.name).all()
+    assert data.index.name == 'number'
+    assert (data.columns == df.columns).all()
+
+
+def test_partitions(db):
+    data = read_sql_table('test', db, columns=list(df.columns), partitions=2,
+                          index_col='number')
+    assert len(data.divisions) == 3
+    assert (data.name.compute() == df.name).all()
+    data = read_sql_table('test', db, columns=['name'], partitions=6,
+                          index_col="number").compute()
+    assert (data.name == df.name).all()
+
+    data = read_sql_table('test', db, columns=['name'], partitions=[0, 2, 4],
+                          index_col="number")
+    assert data.divisions == (0, 2, 4)
+    assert data.index.max().compute() == 3
+
+
+def test_range(db):
+    data = read_sql_table('test', db, partitions=2, index_col='number', limits=[1, 4])
+    assert data.index.min().compute() == 1
+    assert data.index.max().compute() == 3
+
+
+def test_datetimes():
+    import datetime
+    now = datetime.datetime.now()
+    d = datetime.timedelta(seconds=1)
+    df = pd.DataFrame({'a': list('ghjkl'), 'b': [now + i*d for i in range(2, -3, -1)]})
+    with tmpfile() as f:
+        uri = 'sqlite:///%s' % f
+        df.to_sql('test', uri, index=True, if_exists='replace')
+        data = read_sql_table('test', uri, partitions=2, index_col='b')
+        assert data.index.dtype.kind == "M"
+        assert data.divisions[0] == df.b.min()
```

Comment:
It would be good to use assert_eq in here somewhere if possible.  This checks a number of other invariants on the inputs that we expect to hold in dask.dataframe.

    assert_eq(df, data)